R/helper.R

```diff
@@ -19,3 +19,39 @@ get_by_id = function(xs, id) {
   }
   return(NULL)
 }
+
+
+#' @description calculate pareto front of an all-numeric data frame (or matrix)
+#'   also works for >2D pareto front
+#' @example data_frame = data.frame(c(1, 0.25, 0.75, 0), c(0, 0.25, 0.75, 1))
+#'   calculate_pareto_front(data_frame)
+#'   data_frame = data.frame(c(1,1,0,1), c(1,1,0,0), c(1,0,1,0), c(1,1,1,1))
+#'   calculate_pareto_front(data_frame, maximize = FALSE)
+calculate_pareto_front = function(data_frame, maximize = TRUE) {
+
+  assert_data_frame(data_frame, types = "numeric")
```

Comment:
As we are dealing with only numerics, it might be clearer (and probably more efficient in the following) if we convert to a matrix here.